torch/csrc/api/include/torch/serialization.h

```diff
@@ -2,45 +2,69 @@
 
 #include <fstream>
 
-#include <torch/optim.h>
 #include <torch/tensor.h>
 #include <torch/utils.h>
 
+#if defined(TORCH_WITH_CEREAL)
+#include <cereal/access.hpp>
+#include <cereal/cereal.hpp>
+#include <cereal/types/polymorphic.hpp>
+
 #include "cereal/archives/binary.hpp"
-#include "cereal/types/polymorphic.hpp"
 
 #include "cereal/types/string.hpp"
 #include "cereal/types/unordered_map.hpp"
 #include "cereal/types/vector.hpp"
+#endif // defined(TORCH_WITH_CEREAL)
 
 namespace torch {
-
 // Some convenience functions for saving and loading
 template <typename T>
 void save(std::ostream& stream, T const& obj) {
+#if defined(TORCH_WITH_CEREAL)
   cereal::BinaryOutputArchive archive(stream);
   archive(*obj);
+#else
+  AT_ERROR("PyTorch compiled without serialization support");
+#endif
 }
+
 template <typename T>
 void load(std::istream& stream, T& obj) {
+#if defined(TORCH_WITH_CEREAL)
   cereal::BinaryInputArchive archive(stream);
   archive(*obj);
+#else
+  AT_ERROR("PyTorch compiled without serialization support");
+#endif
 }
+
 template <typename T>
 void save(std::ostream& stream, T const* obj) {
+#if defined(TORCH_WITH_CEREAL)
   cereal::BinaryOutputArchive archive(stream);
   archive(*obj);
+#else
+  AT_ERROR("PyTorch compiled without serialization support");
```

Comment:
I'd add here "without serialization support for C++ modules, it's WIP for the unified tensor format"